core/src/test2/java/com/transferwise/common/gracefulshutdown/strategies/TaskSchedulersGracefulShutdownStrategyIntTest.java

```diff
@@ -0,0 +1,81 @@
+package com.transferwise.common.gracefulshutdown.strategies;
+
+import com.transferwise.common.gracefulshutdown.GracefulShutdownHealthIndicator;
+import com.transferwise.common.gracefulshutdown.GracefulShutdownStrategiesRegistry;
+import com.transferwise.common.gracefulshutdown.GracefulShutdowner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.concurrent.ConcurrentTaskScheduler;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles({"test-fast-shutdown"})
+@SpringBootTest(classes = {TaskSchedulersGracefulShutdownStrategyIntTest.TestApplication.class})
+class TaskSchedulersGracefulShutdownStrategyIntTest {
+
+  @SpringBootApplication
+  @EnableScheduling
+  public static class TestApplication {
+
+    @Configuration
+    public static class ExecutorsConfig {
+
+      @Bean
+      ConcurrentTaskScheduler getConcurrentTaskScheduler() {
+        return new ConcurrentTaskScheduler();
+      }
+    }
+  }
+
+  @Autowired
+  private GracefulShutdowner gracefulShutdowner;
+  @Autowired
+  private GracefulShutdownHealthIndicator healthIndicator;
+  @Autowired
+  TaskSchedulersGracefulShutdownStrategy taskSchedulersGracefulShutdownStrategy;
+  @Autowired
+  private GracefulShutdownStrategiesRegistry gracefulShutdownStrategiesRegistry;
+
+  @Autowired
+  private ConcurrentTaskScheduler concurrentTaskScheduler;
+
+  @Test
+  @Disabled("Disabled: Test will fail. Created to show problem with current implementation.")
+  void test_when_task_is_running_longer_than_shutdown_timeout() {
```

Comment:
This test show problem described here: [comment](https://github.com/transferwise/tw-graceful-shutdown/pull/27/files#r1082806844)

Reply:
As discussed this test can be enabled in another PR where we'll fix the existing behvior.